stl/error_formatter.py

```diff
@@ -0,0 +1,100 @@
+import abc
+import json
+import re
+
+
+class ErrorFormatter(object):
+  """Interface for displaying STL parsing errors."""
+  __metaclass__ = abc.ABCMeta
+
+  @abc.abstractmethod
+  def format(self, error):
+    """Outputs parser_error.ErrorInfo |error| in a custom format."""
+    pass
+
+
+class JsonErrorFormatter(ErrorFormatter):
+  """Format errors as json objects."""
+
+  def format(self, error):
+    """Format |error| as json."""
+    return json.dumps(error._asdict())
+
+
+class Color(object):
+  """ANSI escape code base color codes.
+
+  For foreground colors, add 30 to the color value.
+  For background colors, add 40 to the color value.
```

Comment:
Instead of this comment, how about static methods returning fore/background color for given color code.